### PR TITLE
Added better support to paragraphs, lists and dividers

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR is part of the markdown parser work, there's some logic reviewed and now also the formatter is implemented. 

- the list token is now a `ComplexToken` in order to store some information
- handled tokenisation for dotted list
- handled formatting for lists
- handled case of list where text goes to a new line
- added few more tokens to help the parsing
- handled all type of dividers

## Test Plan

Added tests for formatters

### CODE REVIEW

File: packages/@romefrontend/formatter/builders/markdown/blocks/MarkdownHeadingBlock.ts [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 	node: MarkdownHeadingBlock,
 ): Token {
-	throw new Error("unimplemented");
+	const tokens: Tokens = [...Array.from({length: node.level}).map(() => "#")];
+
+	tokens.push(space);
+
+	tokens.push(node.value);
+
+	return concat([...tokens, hardline]);
```
Comment (ooflorent) [inline packages/@romefrontend/formatter/builders/markdown/blocks/MarkdownHeadingBlock.ts:21]:
```suggestion
  return concat([
    "#".repeat(node.level),
    space,
    node.value,
    hardline,
  ]);
```

File: packages/@romefrontend/formatter/builders/markdown/core/MarkdownListItem.ts [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+import {Builder, Token, Tokens, concat} from "@romefrontend/formatter";
 
 export default function MarkdownListItem(
 	builder: Builder,
 	node: MarkdownListItem,
 ): Token {
-	throw new Error("unimplemented");
+	const tokens: Tokens = [];
+	if (node.checked !== undefined) {
+		tokens.push(`[ ${node.checked ? "x" : " "} ]`);
```
Comment (ooflorent) [inline packages/@romefrontend/formatter/builders/markdown/core/MarkdownListItem.ts:10]:
```suggestion
		tokens.push(`[${node.checked ? "x" : " "}]`);
```

Comment (ematipico) [reply packages/@romefrontend/formatter/builders/markdown/core/MarkdownListItem.ts:10]: Strange that our formatter didn't catch it!

**Comment by sebmck** [comment]
Love it! Looks good!

**Pull request merged**